beacon-chain/core/state/transition.go

```diff
@@ -32,25 +32,33 @@ var processDepositsFunc = func(ctx context.Context, s iface.BeaconState, blk *et
 	return b.ProcessDeposits(ctx, s, blk.Block.Body.Deposits)
 }
 var processProposerSlashingFunc = func(ctx context.Context, s iface.BeaconState, blk *ethpb.SignedBeaconBlock) (iface.BeaconState, error) {
-	return b.ProcessProposerSlashings(ctx, s, blk, v.SlashValidator)
```

Comment:
Previously, within processor function `blk` and `blk.Body` was checked for nilness. If there are code paths that depended on that check, we might end up with panics (this method is used for `ProcessBlock()` when going through processing pipeline, no extra checks there either).

NB: The `processDepositsFunc()` (above) doesn't check for block/body nilness, and we seem to be ok. So, if you absolutely sure that check was redundant, and we can have a pipeline items w/o it, please, feel free to dismiss the review. Otherwise, checks should be added to `processAttesterSlashingFunc`, `processEth1DataFunc()`, and `processExitFunc()`

Reply:
Better safe than sorry. I added `VerifyNilBeaconBlock` to `ProcessBlock`. Thanks!